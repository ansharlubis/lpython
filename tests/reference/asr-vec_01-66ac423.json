{
    "basename": "asr-vec_01-66ac423",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/vec_01.py",
    "infile_hash": "eb012561316a4c1f1359bf7b5a0b9aee40944ece197a5bafd387a19f",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-vec_01-66ac423.stdout",
    "stdout_hash": "c9dc305d05386dc7c0dabc2ebfd54004b0fb0e6d919d7bf09eb69779",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}