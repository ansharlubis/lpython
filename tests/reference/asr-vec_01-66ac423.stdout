([1m[35mTranslationUnit[39m[0m 
   ([33mSymbolTable[39m 
      1 
      {
         loop_vec: 
            ([1m[35mSubroutine[39m[0m 
               ([33mSymbolTable[39m 
                  2 
                  {
                     a: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           a [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mReal[39m[0m 8 [
                              (([1m[35mIntegerConstant[39m[0m [36m0[39m 
                                 ([1m[35mInteger[39m[0m 4 [])) 
                                 ([1m[35mIntegerBinOp[39m[0m ([1m[35mIntegerConstant[39m[0m [36m9216[39m 
                                    ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                                    Sub[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                                    ([1m[35mInteger[39m[0m 4 [])) 
                                    ([1m[35mInteger[39m[0m 4 []) ([1m[35mIntegerConstant[39m[0m [36m9215[39m 
                                    ([1m[35mInteger[39m[0m 4 []))))]) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     b: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           b [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mReal[39m[0m 8 [
                              (([1m[35mIntegerConstant[39m[0m [36m0[39m 
                                 ([1m[35mInteger[39m[0m 4 [])) 
                                 ([1m[35mIntegerBinOp[39m[0m ([1m[35mIntegerConstant[39m[0m [36m9216[39m 
                                    ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                                    Sub[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                                    ([1m[35mInteger[39m[0m 4 [])) 
                                    ([1m[35mInteger[39m[0m 4 []) ([1m[35mIntegerConstant[39m[0m [36m9215[39m 
                                    ([1m[35mInteger[39m[0m 4 []))))]) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     i: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           i [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mInteger[39m[0m 4 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.)
                  }) 
               loop_vec [] [
               ([1m[35mDoLoop[39m[0m 
                  (
                     ([1m[35mVar[39m[0m 2 [33mi[39m) ([1m[35mIntegerConstant[39m[0m [36m0[39m 
                     ([1m[35mInteger[39m[0m 4 [])) 
                     ([1m[35mIntegerBinOp[39m[0m ([1m[35mIntegerConstant[39m[0m [36m9216[39m 
                        ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                        Sub[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                        ([1m[35mInteger[39m[0m 4 [])) 
                        ([1m[35mInteger[39m[0m 4 []) ([1m[35mIntegerConstant[39m[0m [36m9215[39m 
                        ([1m[35mInteger[39m[0m 4 []))) ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                     ([1m[35mInteger[39m[0m 4 []))) [
                  ([1m[35m=[39m[0m 
                     ([1m[35mArrayItem[39m[0m 
                        ([1m[35mVar[39m[0m 2 [33mb[39m) [
                        (() 
                           ([1m[35mVar[39m[0m 2 [33mi[39m) ())] 
                        ([1m[35mReal[39m[0m 8 []) ()) 
                     ([1m[35mRealConstant[39m[0m   5.00000000000000000e+00 
                        ([1m[35mReal[39m[0m 8 [])) ())]) 
               ([1m[35mDoLoop[39m[0m 
                  (
                     ([1m[35mVar[39m[0m 2 [33mi[39m) ([1m[35mIntegerConstant[39m[0m [36m0[39m 
                     ([1m[35mInteger[39m[0m 4 [])) 
                     ([1m[35mIntegerBinOp[39m[0m ([1m[35mIntegerConstant[39m[0m [36m9216[39m 
                        ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                        Sub[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                        ([1m[35mInteger[39m[0m 4 [])) 
                        ([1m[35mInteger[39m[0m 4 []) ([1m[35mIntegerConstant[39m[0m [36m9215[39m 
                        ([1m[35mInteger[39m[0m 4 []))) ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                     ([1m[35mInteger[39m[0m 4 []))) [
                  ([1m[35m=[39m[0m 
                     ([1m[35mArrayItem[39m[0m 
                        ([1m[35mVar[39m[0m 2 [33ma[39m) [
                        (() 
                           ([1m[35mVar[39m[0m 2 [33mi[39m) ())] 
                        ([1m[35mReal[39m[0m 8 []) ()) 
                     ([1m[35mArrayItem[39m[0m 
                        ([1m[35mVar[39m[0m 2 [33mb[39m) [
                        (() 
                           ([1m[35mVar[39m[0m 2 [33mi[39m) ())] 
                        ([1m[35mReal[39m[0m 8 []) ()) ())]) 
               ([1m[35mDoLoop[39m[0m 
                  (
                     ([1m[35mVar[39m[0m 2 [33mi[39m) ([1m[35mIntegerConstant[39m[0m [36m0[39m 
                     ([1m[35mInteger[39m[0m 4 [])) 
                     ([1m[35mIntegerBinOp[39m[0m ([1m[35mIntegerConstant[39m[0m [36m9216[39m 
                        ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                        Sub[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                        ([1m[35mInteger[39m[0m 4 [])) 
                        ([1m[35mInteger[39m[0m 4 []) ([1m[35mIntegerConstant[39m[0m [36m9215[39m 
                        ([1m[35mInteger[39m[0m 4 []))) ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                     ([1m[35mInteger[39m[0m 4 []))) [
                  ([1m[35mAssert[39m[0m 
                     ([1m[35mRealCompare[39m[0m 
                        ([1m[35mArrayItem[39m[0m 
                           ([1m[35mVar[39m[0m 2 [33ma[39m) [
                           (() 
                              ([1m[35mVar[39m[0m 2 [33mi[39m) ())] 
                           ([1m[35mReal[39m[0m 8 []) ()) [1m[32m
                        Eq[39m[0m 
                        ([1m[35mRealConstant[39m[0m   5.00000000000000000e+00 
                           ([1m[35mReal[39m[0m 8 [])) 
                        ([1m[35mLogical[39m[0m 4 []) ()) ())])] [1m[32m
               Source[39m[0m [1m[32m
               Public[39m[0m [1m[32m
               Implementation[39m[0m () .false. .false.)
      }) 
   [
   ([1m[35mSubroutineCall[39m[0m 1 [33mloop_vec[39m () [] ())])
(TranslationUnit (SymbolTable 1 {_lpython_main_program: (Subroutine (SymbolTable 4 {}) _lpython_main_program [] [(SubroutineCall 1 loop_vec () [] ())] Source Public Implementation () .false. .false.), loop_vec: (Subroutine (SymbolTable 2 {a: (Variable 2 a Local () () Default (Real 8 [((IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerConstant 9216 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 9215 (Integer 4 []))))]) Source Public Required .false.), b: (Variable 2 b Local () () Default (Real 8 [((IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerConstant 9216 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 9215 (Integer 4 []))))]) Source Public Required .false.), i: (Variable 2 i Local () () Default (Integer 4 []) Source Public Required .false.)}) loop_vec [] [(DoLoop ((Var 2 i) (IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerConstant 9216 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 9215 (Integer 4 []))) (IntegerConstant 1 (Integer 4 []))) [(= (ArrayItem (Var 2 b) [(() (Var 2 i) ())] (Real 8 []) ()) (RealConstant   5.00000000000000000e+00 (Real 8 [])) ())]) (DoLoop ((Var 2 i) (IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerConstant 9216 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 9215 (Integer 4 []))) (IntegerConstant 1 (Integer 4 []))) [(= (ArrayItem (Var 2 a) [(() (Var 2 i) ())] (Real 8 []) ()) (ArrayItem (Var 2 b) [(() (Var 2 i) ())] (Real 8 []) ()) ())]) (DoLoop ((Var 2 i) (IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerConstant 9216 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 9215 (Integer 4 []))) (IntegerConstant 1 (Integer 4 []))) [(Assert (RealCompare (ArrayItem (Var 2 a) [(() (Var 2 i) ())] (Real 8 []) ()) Eq (RealConstant   5.00000000000000000e+00 (Real 8 [])) (Logical 4 []) ()) ())])] Source Public Implementation () .false. .false.), main_program: (Program (SymbolTable 3 {}) main_program [] [(SubroutineCall 1 _lpython_main_program () [] ())])}) [])
